pingora-core/src/upstreams/peer.rs

```diff
@@ -407,9 +408,12 @@ impl HttpPeer {
     }
 
     /// Create a new [`HttpPeer`] with the given path to Unix domain socket and TLS settings.
-    pub fn new_uds(path: &str, tls: bool, sni: String) -> Self {
-        let addr = SocketAddr::Unix(UnixSocketAddr::from_pathname(Path::new(path)).unwrap()); //TODO: handle error
-        Self::new_from_sockaddr(addr, tls, sni)
+    pub fn new_uds(path: &str, tls: bool, sni: String) -> Result<Self, BError> {
```

Comment:
Let's use a `pingora_error::Result` here:

```
 pub fn new_uds(path: &str, tls: bool, sni: String) -> Result<Self> {
```

Then we can use [or_err](https://github.com/cloudflare/pingora/blob/main/pingora-error/src/lib.rs#L515) which will handle the `map_err` and `Error::because` for us:
```
let addr = SocketAddr::Unix(
            UnixSocketAddr::from_pathname(Path::new(path))
                .or_err(ErrorType::InternalError, "invalid path")?,
        );
```